frontend/server/src/Controllers/Problem.php

```diff
@@ -4647,11 +4648,15 @@ public static function getProblemDetailsForTypeScript(
             $r->identity->user_id
         );
 
+        $reviewedData = \OmegaUp\DAO\QualityNominations::getReviewedData(
+            $r->identity,
+            $problem
+        );
+
+        $alreadyReviewed = count($reviewedData) > 0;
+
         $nominationPayload = [
-            'alreadyReviewed' => \OmegaUp\DAO\QualityNominations::reviewerHasQualityTagNominatedProblem(
-                $r->identity,
-                $problem
-            ),
+            'alreadyReviewed' => $alreadyReviewed,
```

Comment:
Como `getReviewedData` pudiera regresar `null`, entonces aquí podrías asignar el valor inline:
```suggestion
            'alreadyReviewed' => !is_null($alreadyReviewed),
```
de esa forma, ya podrías eliminar L4656